src/uct/ib/rc/base/rc_iface.c

```diff
@@ -952,7 +952,7 @@ ucs_status_t uct_rc_iface_qp_create(uct_rc_iface_t *iface, int qp_type,
 #  endif
 
 #if HAVE_IBV_EXP_RES_DOMAIN
-    if (iface->super.res_domain != NULL) {
+    if (iface->super.config.res_domain && (iface->super.res_domain != NULL)) {
```

Comment:
should be enough to check `iface->super.res_domain != NULL`

Reply:
fixed